pxteditor/editor.ts

```diff
@@ -61,9 +62,23 @@ namespace pxt.editor {
         name?: string;
         documentation?: string;
         filesOverride?: pxt.Map<string>;
+        filters?: ProjectFilters;
         temporary?: boolean;
     }
 
+    export interface ProjectFilters {
+        namespaces?: { [index: string]: FilterState; }; // Disabled = 2, Hidden = 0, Visible = 1
+        blocks?: { [index: string]: FilterState; }; // Disabled = 2, Hidden = 0, Visible = 1
+        fns?: { [index: string]: FilterState; }; // Disabled = 2, Hidden = 0, Visible = 1
+        defaultState?: FilterState; // hide, show or disable all by default
+    }
+
+    export enum FilterState {
+        Hidden = 0,
+        Visible = 1,
```

Comment:
Nah I'd say most of the time, you're gonna want to hide everything and only show certain blocks, rather than the other way around.